.github/workflows/platform-environment-matrix.yaml

```diff
@@ -26,12 +30,26 @@ jobs:
         name: Generate Matrix
         run: |
           # generate matrix
-          for f in environments/*/config.yaml ; do
-            if "${{ inputs.include_release }}"; then
-              [ -e "${f}" ] && e+="{\"environment\":\"$(yq .environment "${f}")\",\"platform\":\"$(yq .platform.vendor "${f}")\",\"release\":\"$(yq .release "${f}")\"},"
-            else
-              [ -e "${f}" ] && e+="{\"environment\":\"$(yq .environment "${f}")\",\"platform\":\"$(yq .platform.vendor "${f}")\"},"
-            fi
+          YAML_FILE=environments/environments.yaml 
+          include_all=${{ inputs.include_all }}
+
+          yaml_list={}
+          if [[ ! -f "$YAML_FILE" ]]; then
+            echo "File $YAML_FILE does not exist. Including all."
+            include_all=true
+          else
+              yaml_list=$(yq eval '.enabled[]' "$YAML_FILE")
+          fi
+
+          for f in environments/*/config.yaml; do
+              env=$(echo "$f" | cut -d'/' -f2)
+              if echo "$yaml_list" | grep -wq "^$env$" || [[ "${include_all}" == true ]]; then
+                  if "${include_all}"; then
```

Comment:
This should be `${{ inputs.include_release }}`, also indentation is off in this block